internal/api/sms_provider/twilio_verify.go

```diff
@@ -0,0 +1,132 @@
+package sms_provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/supabase/gotrue/internal/conf"
+	"github.com/supabase/gotrue/internal/utilities"
+)
+
+const (
+	verifyServiceApiBase = "https://verify.twilio.com/v2/Services/"
+)
+
+type TwilioVerifyProvider struct {
+	Config  *conf.TwilioVerifyProviderConfiguration
+	APIPath string
+}
+
+type VerificationResponse struct {
+	To           string `json:"to"`
+	Status       string `json:"status"`
+	Channel      string `json:"channel"`
+	Valid        bool   `json:"valid"`
+	ErrorCode    string `json:"error_code"`
+	ErrorMessage string `json:"error_message"`
+}
+
+// See: https://www.twilio.com/docs/verify/api/verification-check
+type VerificationCheckResponse struct {
+	To           string `json:"to"`
+	Status       string `json:"status"`
+	Channel      string `json:"channel"`
+	Valid        bool   `json:"valid"`
+	ErrorCode    string `json:"error_code"`
+	ErrorMessage string `json:"error_message"`
+}
+
+// Creates a SmsProvider with the Twilio Config
+func NewTwilioVerifyProvider(config conf.TwilioVerifyProviderConfiguration) (SmsProvider, error) {
+	if err := config.Validate(); err != nil {
+		return nil, err
+	}
+	apiPath := verifyServiceApiBase + config.MessageServiceSid + "/Verifications"
+
+	return &TwilioVerifyProvider{
+		Config:  &config,
+		APIPath: apiPath,
+	}, nil
+}
+
+func (t *TwilioVerifyProvider) SendMessage(phone string, message string, channel string) (string, error) {
+	switch channel {
+	case SMSProvider, WhatsappProvider:
+		return "", t.SendSms(phone, message, channel)
+	default:
+		return "", fmt.Errorf("channel type %q is not supported for Twilio", channel)
+	}
+}
+
+// Send an SMS containing the OTP with Twilio's API
+func (t *TwilioVerifyProvider) SendSms(phone, message, channel string) error {
```

Comment:
Pretty sure there's a message ID to be returned here?

Reply:
Hm I don't think there's a MessageID for Twilio Verify. 

Unless you are referring to the Verification ID? https://www.twilio.com/docs/verify/api/verification?code-sample=code-start-a-verification-with-sms&code-language=curl&code-sdk-version=json#verification-response-properties

Reply:
(Wondering because Verifications [don't give an indication of Delivery unlike the MessageID](https://www.twilio.com/docs/verify/api/verification) not sure how useful it will be in terms of debugging)